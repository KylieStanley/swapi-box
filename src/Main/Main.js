import React from 'react';
import CardContainer from '../CardContainer/CardContainer'
import Yoda from '../Yoda/Yoda'
import Menu from '../Menu/Menu'
import { Route, Switch } from 'react-router-dom'

const Main = ({ vehicles, planets, people, favorites, addFavorite}) => {

    return (
      <div className='app'>
        <div class="stars"></div>
        <div class="twinkling"></div>
        <Menu />
      <Switch>
        <Route exact path='/main' component={Yoda}/>
        <Route exact path='/main/people' render={({match}) => <CardContainer data={people} addFavorite={ addFavorite }/> } />
        <Route exact path='/main/vehicles' render={({match}) => <CardContainer data={vehicles} addFavorite={ addFavorite } /> } />
        <Route exact path='/main/planets' render={({match}) => <CardContainer data={planets} addFavorite={ addFavorite } /> } />
        <Route exact path='/main/favorites' render={({match}) => <CardContainer data={favorites} addFavorite={ addFavorite } /> } />
      </Switch>
    </div>
    )

  }

  export default Main

