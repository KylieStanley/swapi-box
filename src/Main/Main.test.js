import React from 'react'
import { shallow } from 'enzyme'
import { Route, component } from 'react-router-dom'
import Main from './Main'

describe('Main', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Main
        vehicles={[]}
        planets={[]}
        people={[]}
        favorites={[]}
        handleFavorite={jest.fn()}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render four card containers for the available categories', () => {
    expect(wrapper.find(Route).length).toEqual(5)
  })

  it('routes /main to the Yoda component', () => {
    expect(wrapper.find('Route[exact=true][path="/main"]').first().prop('render')).toEqual(component)
  })
})
