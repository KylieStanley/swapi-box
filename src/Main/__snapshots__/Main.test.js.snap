// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    favorites={Array []}
    handleFavorite={[MockFunction]}
    people={Array []}
    planets={Array []}
    vehicles={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stars"
        />,
        <div
          className="twinkling"
        />,
        <Menu
          favorites={Array []}
        />,
        <Route
          component={[Function]}
          exact={true}
          path="/main"
        />,
        <Route
          exact={true}
          path="/main/People"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/main/vehicles"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/main/Planets"
          render={[Function]}
        />,
        <Route
          exact={true}
          path="/main/Favorites"
          render={[Function]}
        />,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "twinkling",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "favorites": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": [Function],
          "exact": true,
          "path": "/main",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/main/People",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/main/vehicles",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/main/Planets",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "exact": true,
          "path": "/main/Favorites",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stars"
          />,
          <div
            className="twinkling"
          />,
          <Menu
            favorites={Array []}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/main"
          />,
          <Route
            exact={true}
            path="/main/People"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/main/vehicles"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/main/Planets"
            render={[Function]}
          />,
          <Route
            exact={true}
            path="/main/Favorites"
            render={[Function]}
          />,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "twinkling",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "favorites": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/main",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/main/People",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/main/vehicles",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/main/Planets",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "exact": true,
            "path": "/main/Favorites",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
